src/commands/functions/create.js

```diff
@@ -301,12 +303,50 @@ const downloadFromURL = async function (context, flags, args, functionsDir) {
   }
 }
 
-const installDeps = function (functionPath) {
-  return new Promise((resolve) => {
-    cp.exec('npm i', { cwd: path.join(functionPath) }, () => {
-      resolve()
-    })
-  })
+// Takes an existing list of packages and a list of packages needed by a
+// function, and computes the list of npm modules with version range that
+// must be installed.
+const getNpmInstallPackages = (existingPackages = {}, neededPackages = {}) =>
```

Comment:
Would this work for Yarn users?

Reply:
The current implementation works with npm only. I agree that it should be revisited, but I don't think that belongs in this PR.

Reply:
Makes sense :+1: